srv/salt/_modules/rgw.py

```diff
@@ -104,9 +107,19 @@ def add_users(pathname="/srv/salt/ceph/rgw/cache"):
                 log.info("stderr: {}".format(line))
 
             proc.wait()
-        
-        
-        
+
+
+def create_bucket(**kwargs):
+    s3conn = boto.connect_s3(
+        aws_access_key_id=kwargs['access_key'],
+        aws_secret_access_key=kwargs['secret_key'],
+        host=kwargs['host'],
+        is_secure=False,
+        calling_format=boto.s3.connection.OrdinaryCallingFormat(),
+    )
+    s3conn.create_bucket(kwargs['bucket_name'])
```

Comment:
I'd wrap this in a try catch block. create_bucket would raise a boto.exception.S3CreateError or so... 
http://boto.cloudhackers.com/en/latest/ref/boto.html#boto.exception.S3CreateError